crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2361,10 +2339,11 @@ impl<'db> Type<'db> {
                 //     def __get__(self, instance: object, owner: type | None = None, /) -> MethodType: ...
                 // ```
 
-                #[salsa::tracked(return_ref)]
-                fn overloads<'db>(db: &'db dyn Db) -> CallableSignature<'db> {
```

Comment:
The `signatures` function as a whole is now tracked, instead of tracking each of these individual special case clauses.